pkg/resourcequotas/resourcequotas_test.go

```diff
@@ -0,0 +1,404 @@
+			Definition: &corev1.ResourceQuota{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testRQ",
+					Namespace: "testNamespace",
+				},
+			},
+		}
+
+		if testCase.builderNil {
+			testBuilder = nil
+		}
+
+		if testCase.definitionNil {
+			testBuilder.Definition = nil
+		}
+
+		result := testBuilder.WithQuotaSpec(corev1.ResourceQuotaSpec{
+			Hard: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1"),
+				corev1.ResourceMemory: resource.MustParse("1Gi"),
+			},
+		})
+
+		if !testCase.expectedError {
+			assert.NotNil(t, result)
+			assert.NotNil(t, testBuilder.Definition)
+			assert.Equal(t, resource.MustParse("1"), testBuilder.Definition.Spec.Hard[corev1.ResourceCPU])
+			assert.Equal(t, resource.MustParse("1Gi"), testBuilder.Definition.Spec.Hard[corev1.ResourceMemory])
+		} else {
+			if !testCase.builderNil {
+				assert.NotNil(t, result)
+			} else {
+				assert.Nil(t, result)
+			}
+		}
+	}
+}
+
+func TestResourceQuotaCreate(t *testing.T) {
+	testCases := []struct {
+		rqExistsAlready bool
+	}{
+		{
+			rqExistsAlready: true,
+		},
+		{
+			rqExistsAlready: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		testRQ := generateResourceQuota("testRQ", "testNamespace")
+
+		if testCase.rqExistsAlready {
+			runtimeObjects = append(runtimeObjects, testRQ)
+		}
+
+		testBuilder := buildTestBuilderWithFakeObjects(runtimeObjects, testRQ.Name, testRQ.Namespace)
+		builderResult, err := testBuilder.Create()
+
+		assert.Nil(t, err)
+		assert.NotNil(t, testBuilder.Object)
+		assert.Equal(t, testRQ.Name, builderResult.Object.Name)
+		assert.Equal(t, testRQ.Namespace, builderResult.Object.Namespace)
+	}
+}
+
+func TestResourceQuotaDelete(t *testing.T) {
+	testCases := []struct {
+		rqExistsAlready bool
+	}{
+		{
+			rqExistsAlready: true,
+		},
+		{
+			rqExistsAlready: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		testRQ := generateResourceQuota("testRQ", "testNamespace")
+
+		if testCase.rqExistsAlready {
+			runtimeObjects = append(runtimeObjects, testRQ)
+		}
+
+		testBuilder := buildTestBuilderWithFakeObjects(runtimeObjects, testRQ.Name, testRQ.Namespace)
+		err := testBuilder.Delete()
+
+		assert.Nil(t, err)
+		assert.Nil(t, testBuilder.Object)
+	}
+}
+
+func TestResourceQuotaExists(t *testing.T) {
+	testCases := []struct {
+		rqExistsAlready bool
+	}{
+		{
+			rqExistsAlready: true,
+		},
+		{
+			rqExistsAlready: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+
+		testRQ := generateResourceQuota("testRQ", "testNamespace")
+
+		if testCase.rqExistsAlready {
+			runtimeObjects = append(runtimeObjects, testRQ)
+		}
+
+		testBuilder := buildTestBuilderWithFakeObjects(runtimeObjects, testRQ.Name, testRQ.Namespace)
+		result := testBuilder.Exists()
+
+		assert.Equal(t, testCase.rqExistsAlready, result)
+	}
+}
+
+func TestNewBuilder(t *testing.T) {
```

Comment:
Please move this function of top of the file and add TestResourceQuota* Prefix

Reply:
Done.